### ISSUE TITLE
Parallel requests with WinHttpHandler throws exception "The server returned an invalid or unrecognized response"

### ISSUE THREAD

**Issue by dpypin** [dotnet/corefx#22195]
According to different recommendations I use only one instance of HttpClient in my app. I need to make HTTP/2 calls, so I use WinHttpHandler. I noticed that when I run several requests simultaneously some of them fail with

> System.Net.Http.HttpRequestException: An error occurred while sending the request. ---> System.Net.Http.WinHttpException: The server returned an invalid or unrecognized response
>    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>    at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
>    at System.Net.Http.WinHttpHandler.<StartRequest>d__105.MoveNext()
>    --- End of inner exception stack trace ---
>    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>    at System.Net.Http.HttpClient.<FinishSendAsync>d__58.MoveNext()
> --- End of stack trace from previous location where exception was thrown ---
>    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>    at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
>    at WinHttpRaceCondition.Program.<SendToApple>d__2.MoveNext() in D:\dev\test-projects\WinHttpRaceCondition\WinHttpRaceCondition\Program.cs:line 76

The following code reproduces this issue.

```
using System;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace WinHttpRaceCondition
{
    using System.Net.Http;
    using System.Security.Authentication;

    class Program
    {
        private static HttpClient _httpClient;

        static void Main(string[] args)
        {
            var httpHandler = new WinHttpHandler { SslProtocols = SslProtocols.Tls12 };
            _httpClient = new HttpClient(httpHandler, true);

            var numberOfRequests = 5;

            if (args.Length == 2)
            {
                var sharedClient = args[0] == "SharedClient=true";
                var parallelExecution = args[1] == "ParallelExecution=true";

                if (parallelExecution)
                {
                    var tasks = new Task<string>[numberOfRequests];

                    for (var i = 0; i < numberOfRequests; i++)
                    {
                        tasks[i] = SendToApple(sharedClient, i.ToString());
                    }

                    Task.WaitAll(tasks);

                    for (var i = 0; i < numberOfRequests; i++)
                    {
                        Console.WriteLine(tasks[i].Result + "\n");
                    }
                }
                else
                {
                    for (var i = 0; i < numberOfRequests; i++)
                    {
                        Console.WriteLine(SendToApple(sharedClient, i.ToString()).Result + "\n");
                    }
                } 
            }
            else
            {
                Console.WriteLine("Wrong arguments");
            }
        }

        private static async Task<string> SendToApple(bool shareHttpClient, string postfix)
        {
            HttpClient httpClient;

            if (shareHttpClient)
            {
                httpClient = _httpClient;
            }
            else
            {
                var httpHandler = new WinHttpHandler { SslProtocols = SslProtocols.Tls12 };
                httpClient = new HttpClient(httpHandler, true);
            }

            using (var request = new HttpRequestMessage(HttpMethod.Post, "https://api.push.apple.com:443/3/device/" + postfix))
            {
                request.Content = new StringContent("Test"); ;
                request.Version = new Version(2, 0);
                try
                {
                    using (var httpResponseMessage = await httpClient.SendAsync(request))
                    {
                        var responseContent = await httpResponseMessage.Content.ReadAsStringAsync();
                        return $"status: {(int) httpResponseMessage.StatusCode} content: {responseContent}";
                    }
                }
                catch (Exception e)
                {
                    return e.ToString();
                }
            }
        }
    }
}
```

SharedClient parameter controls if requests will use shared (true) or separate httpClients (false), ParallelExecution controls if requests will be executed in parallel (true) or sequentually (true).

If you run it with SharedClient=true ParallelExecution=true the program will use shared httpClient for requests and make 5 requests in parallel. In that case it is very probable that a few of these requests fail with "System.Net.Http.WinHttpException: The server returned an invalid or unrecognized response" exception. The program with other combinations of parameters works fine.

**Comment by dpypin** [comment]
I can't reproduce this issue anymore
